modules/metrics-probe/metrics-probe.c

```diff
@@ -128,12 +128,13 @@ _lookup_stats_counter(MetricsProbe *self, LogMessage *msg)
   if (!cluster)
     {
       cluster = _register_single_cluster_locked(&key);
-      g_hash_table_insert(clusters, &cluster->key, cluster);
+      if (cluster)
+        g_hash_table_insert(clusters, &cluster->key, cluster);
     }
 
   scratch_buffers_reclaim_marked(marker);
 
-  return stats_cluster_single_get_counter(cluster);
+  return cluster ? stats_cluster_single_get_counter(cluster) : NULL;
```

Comment:
Shouldn't we make `stats_cluster_single_get_counter()` NULL-safe?
(For example, the `stats-counter.c` unit is completely NULL-safe)

Reply:
Yes, the user side got a lot cleaner, when I added a `NULL` check to `stats_cluster_single_get_counter()`. Thanks!